pennylane/templates/tensornetworks/ttn.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the TTN template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import warnings
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+
+
+def compute_indices(wires, n_block_wires):
+    """Generate a list containing the wires for each block.
+
+    Args:
+        wires (Iterable): wires that the template acts on
+        n_block_wires (int): number of wires per block
+
+    Returns:
+        layers (array): array of wire indices or wire labels for each block
+    """
+
+    n_wires = len(wires)
+
+    if n_block_wires % 2 != 0:
+        raise ValueError(f"n_block_wires must be an even integer; got {n_block_wires}")
+
+    if n_block_wires < 2:
+        raise ValueError(
+            f"number of wires in each block must be larger than or equal to 2; got n_block_wires = {n_block_wires}"
+        )
+
+    if n_block_wires > n_wires:
+        raise ValueError(
+            f"n_block_wires must be smaller than or equal to the number of wires; "
+            f"got n_block_wires = {n_block_wires} and number of wires = {n_wires}"
+        )
+
+    if not np.log2(n_wires / n_block_wires).is_integer():
+        warnings.warn(
+            f"The number of wires should be n_block_wires times 2^n; got n_wires/n_block_wires = {n_wires/n_block_wires}"
+        )
+
+    n_wires = 2 ** (int(np.log2(len(wires) / n_block_wires))) * n_block_wires
```

Comment:
I'm confused. Seems like you are computing `2^(log2(n/m)) = n/m`. So why not just `n/m`? Am I missing something?

Reply:
`2^int(log2(n/m)) = 2^floor(log2(n/m)) <= n/m`
This takes care of the case when we have e.g. 3 wires, 2 wires per block so we want to use only the first 2 wires.
Then `n/m = 1.5` but `2^int(log2(n/m))=2`
At first glance, it will look like we can use something simpler, like `n//m*m`. But for cases like 13 wires, 4 wires per block, we get `n//m*m = 13//4*4 = 12`. Here, `12` is will not give a viable TTN. For a TTN of block size 4, we can use `[4, 8, 16, 32, etc.]` wires. The next step up would be 16 wires. So we have to use the logarithm.
We can discuss this more if it's not clear.

Reply:
Got it, thanks!